evalml/tests/data_checks_tests/test_id_columns_data_check.py

```diff
@@ -168,6 +141,194 @@ def test_id_columns_strings():
         ).to_dict(),
     ]
 
+
+def test_id_cols_data_check_input_formats():
```

Comment:
This test has a lot of repeated code. You can clean this up by parameterizing over the data type and setting up the input data that way - see [here](https://github.com/alteryx/evalml/blob/main/evalml/tests/automl_tests/test_default_algorithm.py#L932) for an example